modules/server/src/mapping/masking.ts

```diff
@@ -0,0 +1,95 @@
+import { Aggregation } from './types';
+
+/**
+ * This returns a total count that is less than or equal to the actual total hits in the query.
+ * It is calculated by adding +1 for values under threshold and bucket.doc_count
+ * for values greater than or equal to
+ *
+ * @param aggregation an aggregation with the most buckets which has data masking applied
+ * @returns hits total value
+ */
+const calculateHitsFromAggregation = ({
+	aggregation,
+}: {
+	aggregation: Aggregation | undefined;
+}) => {
+	if (!aggregation) {
+		console.error('No aggregation found for calculating hits.');
+		return 0;
+	}
+	return aggregation.buckets.reduce(
+		(totalAcc, bucket) => (bucket.belowThreshold ? totalAcc + 1 : totalAcc + bucket.doc_count),
+		0,
+	);
+};
+
+/**
+ *
+ * 1) Iterate through aggs applying data masking to buckets if applicable
+ * 2) Find the agg with the most bucket count and data masking applied to be used in calculating hits.total
+ *
+ * @param aggregations - aggregations from query
+ * @param thresholdMin - threshold value
+ * @returns aggregations with data masking applied and hits total
+ */
+export const applyAggregationMasking = ({
+	aggregations,
+	thresholdMin,
+}: {
+	aggregations: Record<
+		string,
+		{
+			bucket_count: number;
+			buckets: Array<{
+				doc_count: number;
+				key: string;
+			}>;
+		}
+	>;
+	thresholdMin: number;
+}) => {
```

Comment:
Very helpful to see the return type declaration. This one is quite large though and may benefit from using some type aliases. No immediate suggestions from me, but the content of aggregations is probably used in multiple places and a good candidate.

Somethign like (naming only very quickly considered):
```ts
type AggregationData = {
    bucket_count: number;
    buckets: Array<{
    doc_count: number;
    key: string;
  }>;
}

const applyAggregationMasking(...): { aggregations: Record<string, AggregationData>, thresholdMin: number } => {...}
```